presto-spi/src/main/java/com/facebook/presto/spi/block/MapBlock.java

```diff
@@ -155,4 +158,38 @@ public String toString()
         sb.append('}');
         return sb.toString();
     }
+
+    public static MapBlock fromKeyValueBlock(MapType mapType, boolean[] mapIsNull, int[] offsets, Block keyBlock, Block valueBlock)
```

Comment:
In `testWith` in `TestMapBlock`, I extended them so that they also test blocks constructed through this method.